### ISSUE TITLE
vault_kv_secret_v2 incorrect metadata path

### ISSUE THREAD

**Issue by Tonkari** [hashicorp/terraform-provider-vault#1770]
Since upgrading to provider version 3.13.0, our terraform pipeline can no longer update secrets in our kvv2 secret engine. Permissions did not change and a provider downgrade fixed the issue for now.

### Terraform Version
provider registry.terraform.io/hashicorp/vault v3.13.0

### Affected Resource(s)
- vault_kv_secret_v2

### Terraform Configuration Files
```hcl
resource "vault_kv_secret_v2" "credentials" {
  for_each            = toset(var.our-list-of-data)
  mount               = "our-prefix/our-secret-engine"
  name                = each.key
  cas                 = 1
  delete_all_versions = true
  data_json           = jsonencode(our-data[each.key])
}

```

### Error
```
Error: Error making API request.
│ 
│ URL: GET https://redacted.url/v1/our-prefix/metadata/data/our-secret-name
│ Code: 403. Errors:
│ 
│ * 1 error occurred:
│ 	* permission denied
│ 
```

### Expected Behavior
Terraform should be able to update the resource as with provider version 3.12.0

### Actual Behavior
The above error message appears. Note that the path does not contain the name of our secret engine but "metadata/data", so the API call seems to have been constructed incorrectly.

### Steps to Reproduce
1. Create a secret engine with a prefix in its path (for example "prefix/kvv2" instead of just "kvv2")
2. Create a token with permissions to manage secrets in this engine
3. Use the token to create a vault_kv_secret_v2 in this engine

### Important Factoids
The secret engine must contain a "/" in its path, see example above

### References
May have been introduced by https://github.com/hashicorp/terraform-provider-vault/pull/1722

**Comment by sandangel** [comment]
I have the same issue. Switching back to 3.12.0 work for me.

**Comment by sandangel** [comment]
actually I just tried again, it doesn't work. 😢

**Comment by rjhornsby** [comment]
As noted in #1780 (duplicate of this), this issue is nearly identical to the problem outlined in #1186. This might suggest a more fundamental misunderstanding of how vault paths are supposed to be treated and how they can be broken down. #1186 demonstrates how the azure logic assumes the same - seemingly that a vault path can always be split into 3 parts by `/` without considering that an engine mount point might not be located at the root or entirely within the first path component.

**Comment by vinay-gopalan** [comment]
Thanks for reporting this issue! The fix in #1781 has been merged and is set to be released in a patch release v3.13.1 this week. Please do let us know if you see any further issues with this resource! 😄

**Comment by vinay-gopalan** [comment]
Update: we had to move some things around for the release and include some features that were needed after Vault 1.13 was released. This fix has been incorporated as part of the [v3.14.0](https://github.com/hashicorp/terraform-provider-vault/releases/tag/v3.14.0) release

**Comment by revanthalampally** [comment]
@vinay-gopalan : Still seeing the error with Terraform v1.4.4 and provider registry.terraform.io/hashicorp/vault v3.15.0
`Error: expected "auth/pod/dc709249/config" to have 4 pieces, has 4 `
Please let us know when this can be fixed?
